File: OBJCSample/OBJCSample/classes/AZConvertorCalendar.m
//
//  AZConvertorCalendar.m
//  AZCalendarOBJC
//
//  Created by Ali on 4/14/17.
//  Copyright © 2017 Ali Azadeh. All rights reserved.
//

#import "AZConvertorCalendar.h"

@implementation AZConvertorCalendar

+(DateType*)convertDateWithinputType:(CalendarType)inputType inputDate:(DateType*)inputDate outputType:(CalendarType)outputType{

    
    DateType *result = [[DateType alloc] initWithDay:0 month:0 year:0];

    NSCalendarIdentifier input = NSCalendarIdentifierPersian;
    NSCalendarIdentifier output = NSCalendarIdentifierPersian;
    
    
    
    switch (inputType) {
        case persian:
            input = NSCalendarIdentifierPersian;
            break;
        
        case gregorian:
             input = NSCalendarIdentifierGregorian;
            break;
            
        case hijri:
             input = NSCalendarIdentifierIslamicCivil;
            break;
            
        default:
            break;
    }
 
    
    switch (outputType) {
        case persian:
            output = NSCalendarIdentifierPersian;
            break;
            
        case gregorian:
            output = NSCalendarIdentifierGregorian;
            break;
            
        case hijri:
            output = NSCalendarIdentifierIslamicCivil;
            break;
            
        default:
            break;
    }
    
    
    NSCalendar *sourceCal = [[NSCalendar alloc] initWithCalendarIdentifier:input];
    
    NSDateComponents *sourceComponent = [[NSDateComponents alloc] init];
    
    sourceComponent.day = inputDate.day;
    sourceComponent.month = inputDate.month;
    sourceComponent.year = inputDate.year;

    NSDate *sourceDate = [sourceCal dateFromComponents:sourceComponent];
    NSCalendar *destinationCal = [[NSCalendar alloc] initWithCalendarIdentifier:output];
  
    
      NSDateComponents *destinationComponents = [destinationCal components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:sourceDate];
    
    result.day = destinationComponents.day;
    result.month = destinationComponents.month;
    result.year = destinationComponents.year;

    return result;
}

@end

File: SwiftSample/SwiftSample/classes/AZConvertorCalendar.swift
//
//  AZConvertorCalendar.swift
//  AZCalendarSwift
//
//  Created by Ali on 4/11/17.
//  Copyright © 2017 Ali Azadeh. All rights reserved.
//

import Foundation

public enum CalendarType {
    case persian
    case hijri
    case gregorian
}

public class AZConvertorCalendar {
    
    static func convertDate(inputType : CalendarType , inputDate : DateType , outputType : CalendarType)->DateType {
        
        var input = Calendar.Identifier.persian
        var output = Calendar.Identifier.persian
        
        let result = DateType.init(day: 0, month: 0, year: 0)
        
        switch inputType {
        case .persian:
            input = .persian
            break
        case .gregorian :
            input = .gregorian
            break
        case .hijri :
            input = .islamicCivil
            break
        }
        
        switch outputType {
        case .persian:
            output = .persian
            break
        case .gregorian :
            output = .gregorian
            break
        case .hijri :
            output = .islamicCivil
            break
        }
        
        let sourceCal = Calendar.init(identifier: input)
        
        var sourceComponent = DateComponents.init()
        
        sourceComponent.day = inputDate.day
        sourceComponent.month = inputDate.month
        sourceComponent.year = inputDate.year
        
        let sourceDate = sourceCal.date(from: sourceComponent)
        let destinationCal = Calendar.init(identifier: output)
        let destinationComponents = destinationCal.dateComponents([.year, .month, .day], from: sourceDate!)
        result.day = destinationComponents.day!
        result.month = destinationComponents.month!
        result.year = destinationComponents.year!
        
        return result
    }
    
}
